packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmAnyTests.kt

```diff
@@ -61,59 +100,651 @@ class RealmAnyTests {
+                String::class -> loopSupportedTypes(createManagedContainer())
+                Float::class -> loopSupportedTypes(createManagedContainer())
+                Double::class -> loopSupportedTypes(createManagedContainer())
+                BsonObjectId::class -> loopSupportedTypes(createManagedContainer())
+                ByteArray::class -> loopSupportedTypes(createManagedContainer())
+                RealmInstant::class -> loopSupportedTypes(createManagedContainer())
+                RealmUUID::class -> loopSupportedTypes(createManagedContainer())
+                Sample::class -> loopSupportedTypes(createManagedContainer())
+            }
+        }
+    }
+
+    @Test
+    fun managed_coreIntValuesAreTheSame() {
+        assertCoreIntValuesAreTheSame(
+            fromLong = assertNotNull(createManagedRealmAny { RealmAny.create(42) }),
+            fromInt = assertNotNull(createManagedRealmAny { RealmAny.create(42L) }),
+            fromChar = assertNotNull(createManagedRealmAny { RealmAny.create(42.toShort()) }),
+            fromShort = assertNotNull(createManagedRealmAny { RealmAny.create(42.toByte()) }),
+            fromByte = assertNotNull(createManagedRealmAny { RealmAny.create(42.toChar()) })
+        )
+    }
+
+    @Test
+    fun managed_numericOverflow() {
+        assertNumericOverflow {
+            assertNotNull(createManagedRealmAny { it })
+        }
+    }
+
+    @Test
+    fun managed_deletedObject() {
+        val managedContainer = realm.writeBlocking {
+            val unmanagedContainer = RealmAnyContainer(RealmAny.create(Sample()))
+            copyToRealm(unmanagedContainer)
+        }
+        realm.writeBlocking {
+            delete(query<Sample>())
+        }
+        realm.writeBlocking {
+            val updatedContainer = findLatest(managedContainer)
+            assertNull(assertNotNull(updatedContainer).anyField)
+        }
+    }
+
+    @Test
+    fun managed_deleteObjectInsideRealmAnyTriggersUpdateInContainer() {
+        runBlocking {
+            val sampleChannel = Channel<SingleQueryChange<Sample>>(1)
+            val containerChannel = Channel<SingleQueryChange<RealmAnyContainer>>(1)
+
+            val sampleObserver = async {
+                realm.query<Sample>()
+                    .first()
+                    .asFlow()
+                    .collect {
+                        if (it is DeletedObject<Sample>) {
+                            sampleChannel.trySend(it)
+                        }
+                    }
+            }
+            val containerObserver = async {
+                realm.query<RealmAnyContainer>()
+                    .first()
+                    .asFlow()
+                    .collect {
+                        if (it is UpdatedObject<RealmAnyContainer>) {
+                            containerChannel.trySend(it)
+                        }
+                    }
+            }
+
+            val deletionJob = async {
```

Comment:
Hmm, `async` and `delay` ... feels like we should rather trigger the update based on some specific event received from the `sampleChannel` 🤔

Reply:
Yeah, I just ignored the first two events and went straight to the deletion updates 🙈 I'll change it to process all events and be async-failproof.